src/ipc/ipc4/helper.c

```diff
@@ -764,7 +764,7 @@ int ipc4_trigger_chain_dma(struct ipc *ipc, struct ipc4_chain_dma *cdma)
 			}
 		}
 
-		return IPC4_SUCCESS;
+		return ret;
```

Comment:
sorry, don't understand, I don't see any path where a non-success return code would be lost? It looks like everywhere `ret` is in the "negative POSIX or 0" error space, and this seems to mix it with IPC4_* error codes. Yet another one...